File: 88. Merge Sorted Array/solution.py
class Solution(object):
  def merge(self, nums1, m, nums2, n):
    """
    :type nums1: List[int]
    :type m: int
    :type nums2: List[int]
    :type n: int
    :rtype: None Do not return anything, modify nums1 in-place instead.
    """
    nums1Index = m - 1
    nums2Index = n - 1
    insertIndex = m + n - 1

    while insertIndex >= 0 and nums2Index >= 0:
      if nums1Index >= 0 and nums1[nums1Index] > nums2[nums2Index]:
        nums1[insertIndex] = nums1[nums1Index]
        nums1Index -= 1
      else:
        nums1[insertIndex] = nums2[nums2Index]
        nums2Index -= 1
      insertIndex -= 1
    return nums1

File: 88. Merge Sorted Array/solution.java
class Solution {
  public void merge(int[] nums1, int m, int[] nums2, int n) {
    int nums1Index = m - 1;
    int nums2Index = n - 1;
    int insertIndex = m + n - 1;

    while (insertIndex >= 0 && nums2Index >= 0){
      if (nums1Index >= 0 && nums1[nums1Index] > nums2[nums2Index]){
        nums1[insertIndex] = nums1[nums1Index];
        nums1Index -= 1;
      }
      else{
        nums1[insertIndex] = nums2[nums2Index];
        nums2Index -= 1;
      }
      insertIndex -= 1;
    }
  }
}